tests/src/test/java/io/streamnative/pulsar/handlers/kop/SaslOAuthKopHandlersWithGroupIdTest.java

```diff
@@ -0,0 +1,203 @@
+                adminCredentialPath, ISSUER_URL, AUDIENCE));
+        conf.setKafkaEnableAuthorizationForceGroupIdCheck(true);
+        final Properties properties = new Properties();
+        properties.setProperty("tokenPublicKey", tokenPublicKey);
+        conf.setProperties(properties);
+
+        // KoP's config
+        conf.setSaslAllowedMechanisms(Sets.newHashSet("OAUTHBEARER"));
+        conf.setKopOauth2AuthenticateCallbackHandler(OauthValidatorCallbackHandler.class.getName());
+        conf.setKopOauth2ConfigFile("src/test/resources/kop-handler-oauth2.properties");
+
+        super.internalSetup();
+
+        admin.tenants().createTenant(tenant,
+                TenantInfo.builder()
+                        .adminRoles(Collections.singleton(ADMIN_USER))
+                        .allowedClusters(Collections.singleton(configClusterName))
+                        .build());
+        TenantInfo tenantInfo = admin.tenants().getTenantInfo(tenant);
+        log.info("TenantInfo for {} {} in test", tenant, tenantInfo);
+        assertNotNull(tenantInfo);
+    }
+
+    @AfterClass
+    @Override
+    protected void cleanup() throws Exception {
+        super.internalCleanup();
+    }
+
+    @Override
+    protected void createAdmin() throws Exception {
+        super.admin = PulsarAdmin.builder()
+                .serviceHttpUrl(brokerUrl.toString())
+                .authentication(
+                        AuthenticationFactoryOAuth2.clientCredentials(
+                                new URL(ISSUER_URL), new URL(adminCredentialPath), AUDIENCE))
+                .build();
+    }
+
+    @Test(timeOut = 30000)
+    public void testGrantAndRevokePermissionWithGroupId() throws Exception {
+        SaslOAuthKopHandlersTest.OAuthMockAuthorizationProvider.NULL_ROLE_STACKS.clear();
+
+        final String namespace = tenant + "/" + "test-grant-and-revoke-permission-with-group-id-ns";
+        admin.namespaces().createNamespace(namespace);
+        final String topic = "persistent://" + namespace + "/test-grant-and-revoke-permission-with-group-id";
+        final String role = "normal-role-" + System.currentTimeMillis();
+        final String clientCredentialPath = HydraOAuthUtils.createOAuthClient(role, "secret", tenant);
+
+        admin.namespaces().grantPermissionOnNamespace(namespace, role, Collections.singleton(AuthAction.produce));
+
+        final Properties consumerProps = newKafkaConsumerProperties();
+        internalConfigureOAuth2(consumerProps, clientCredentialPath);
+        final KafkaConsumer<String, String> consumer = new KafkaConsumer<>(consumerProps);
+        consumer.subscribe(Collections.singleton(topic));
+
+        admin.namespaces().grantPermissionOnNamespace(namespace, role, Sets.newHashSet(AuthAction.consume));
+        // Only have consume permission, can't consume without subscription permission.
+        Assert.assertThrows(GroupAuthorizationException.class, () -> consumer.poll(Duration.ofSeconds(5)));
```

Comment:
Could you verify why the `TopicAuthorizationException` is thrown here?

```
 SaslOAuthKopHandlersWithGroupIdTest.testGrantAndRevokePermissionWithGroupId:131 Expected GroupAuthorizationException to be thrown, but TopicAuthorizationException was thrown
```

Reply:
Fixed